internal_api.go

```diff
@@ -43,6 +43,41 @@ func (e *Enforcer) addPolicy(sec string, ptype string, rule []string) (bool, err
 	return ruleAdded, nil
 }
 
+// addPolicies adds rules to the current policy.
+// removePolicies removes rules from the current policy.
+func (e *Enforcer) addPolicies(secs []string, ptypes []string, rules [][]string) ([]bool, []error) {
+	rulesAdded := e.model.AddPolicies(secs, ptypes, rules)
+	errs := make([]error, len(rulesAdded))
+	for i, isRuleAdded := range rulesAdded {
+		if !isRuleAdded {
+			errs[i] = nil
+		}
+	}
+
+	if e.adapter != nil && e.autoSave {
+		newErrs := e.adapter.AddPolicies(secs, ptypes, rules)
+		for i, err := range newErrs {
+			if err != nil {
+				if err.Error() != notImplemented {
+					errs[i] = err
+				}
+			}
+		}
+		return rulesAdded, errs
+	}
+
+	if e.watcher !=nil && e.autoNotifyWatcher {
+		err := e.watcher.Update()
```

Comment:
Please move the `watcher` to `for` outside.

Reply:
Sorry, didn't got it? Can you come again. Thank you

Reply:
It means that after all the policy have been added then call the `e.watcher.update()`

Reply:
Yeah, thanks @nodece , its exactly what I have done.